api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -508,7 +508,18 @@ default void setScalar(NDIndex index, Number value) {
      * @return the partial {@code NDArray}
      */
     default NDArray get(NDIndex index) {
-        return getNDArrayInternal().getIndexer().get(this, index);
+        return get(index, getManager());
+    }
+
+    /**
+     * Returns a partial {@code NDArray}.
+     *
+     * @param manager the manager used to create the arrays
+     * @param index the section of this {@code NDArray} to return
+     * @return the partial {@code NDArray}
+     */
+    default NDArray get(NDIndex index, NDManager manager) {
```

Comment:
Minor thing, but I feel like it would be more consistent to put manager as the first argument